.editorconfig

```diff
@@ -17,3 +17,16 @@ dotnet_sort_system_directives_first = true
 
 dotnet_diagnostic.IDE0073.severity = warning  # Require file header
 file_header_template = Copyright (c) Duende Software. All rights reserved.\nSee LICENSE in the project root for license information.
+
+dotnet_diagnostic.IDE0161.severity = error  # Use file-scoped namespace
+csharp_style_namespace_declarations = file_scoped:error
+
+dotnet_diagnostic.IDE0003.severity = error # Enforce that 'this.' should not be used when it's redundant
+
+dotnet_diagnostic.IDE0049.severity = error # Enforce using built-in type keywords (e.g., 'string' instead of 'String')
+
+csharp_new_line_before_open_brace = all # BSD/Allman braces: Place braces on a new line for all constructs
+dotnet_diagnostic.IDE0011.severity = error # Require braces for all if statements
+dotnet_diagnostic.IDE0055.severity = error # Keep empty braces { } on the same line
+csharp_style_allow_blank_line_after_open_brace = false
+csharp_style_allow_blank_line_between_consecutive_braces = false
```

Comment:
- We already have IDE0055 above, so it shouldn't be repeated here.
- `csharp_new_line_before_open_brace` is an option for IDE0055 so it should appear below it
- The title of IDE011 is "Add braces". Also it should be placed between IDE0005 and IDE0055 above to keep the diagnostics in order.
- Google returns no results for `csharp_style_allow_blank_line_after_open_brace` nor `csharp_style_allow_blank_line_between_consecutive_braces`. Where did you find them?

Reply:
Going to redo this, there a lot more formatting option around braces to consider/add.
>csharp_style_allow_blank_line_after_open_brace  pasta